draft-ietf-quic-transport.md

```diff
@@ -2675,201 +2676,289 @@ created, it MUST close the connection with error code STREAM_STATE_ERROR.
+{{fig-stream-recv-states}} shows the states for the part of a stream that
+receives data from a peer.  The states for a receive stream mirror only some the
+states of the send stream at the peer.  A receive stream doesn't track states on
+the send stream that can't be observed, such as the "Open" state; instead,
+receive streams track the delivery of data to the application or application
+protocol.
 
-If the application resets a stream that is already in the "closed" state, a
-RST_STREAM frame MAY still be sent in order to cancel retransmissions of
-previously-sent STREAM frames.
+~~~
+       o
+       | Recv STREAM / STREAM_BLOCKED / RST_STREAM
+       | Open Paired Stream (bidirectional)
+       | Recv MAX_STREAM_DATA
+       v
+   +-------+
+   | Recv  | Recv RST_STREAM
+   |       |-----------------------.
+   +-------+                       |
+       |                           |
+       | Recv STREAM + FIN         |
+       v                           |
+   +-------+                       |
+   | Size  | Recv RST_STREAM       |
+   | Known +---------------------->|
+   +-------+                       |
+       |                           |
+       | Recv All Data             |
+       v                           v
+   +-------+                   +-------+
+   | Data  | Recv RST_STREAM   | Reset |
+   | Recvd +<-- (optional) --->| Recvd |
+   +-------+                   +-------+
+       |                           |
+       | App Read All Data         | App Read RST
+       v                           v
+   +-------+                   +-------+
+   | Data  |                   | Reset |
+   | Read  |                   | Read  |
+   +-------+                   +-------+
+~~~
+{: #fig-stream-recv-states title="States for Receive Streams"}
+
+The receiving part of a stream initiated by a peer (types 1 and 3 for a client,
+or 0 and 2 for a server) are created when the first STREAM, STREAM_BLOCKED, or
+RST_STREAM is received for that stream.  The initial state for a receive stream
+is "Recv".  Receiving a RST_STREAM frame causes the receive stream to
+immediately transition to the "Reset Recvd".
+
+The receive stream enters the "Recv" state when the sending part of a
+bidirectional stream initiated by the endpoint (type 0 for a client, type 1 for
+a server) enters the "Open" state.
+
+A bidirectional stream also opens when a MAX_STREAM_DATA frame is received.
+Receiving a MAX_STREAM_DATA frame implies that the remote peer has opened the
+stream and is providing flow control credit.  A MAX_STREAM_DATA frame might
+arrive before a STREAM or STREAM_BLOCKED frame if packets are lost or reordered.
+
+In the "Recv" state, the endpoint receives STREAM and STREAM_BLOCKED frames.
+Incoming data is buffered and reassembled into the correct order for delivery to
+the application.  As data is consumed by the application and buffer space
+becomes available, the endpoint sends MAX_STREAM_DATA frames to allow the peer
```

Comment:
MAY send?

Reply:
Probably not, honestly - it's a MUST in a sense as well.  Best to leave the normative language about that to other parts of the spec.